src/coprocessor/statistics/analyze.rs

```diff
@@ -297,6 +329,26 @@ impl<S: Snapshot> SampleBuilder<S> {
                 columns_info = &columns_info[1..];
             }
 
+            if self.analyze_common_handle {
+                for logical_row in &result.logical_rows {
+                    let mut data = vec![];
+                    for handle_id in &self.common_handle_col_ids {
+                        let mut handle_col_val = vec![];
```

Comment:
I would recommend use `Vec::with_capacity` if we can know vector length in advance.

Reply:
It's hard to estimate the length.